cranelift/filetests/src/test_run_zkasm.rs

```diff
@@ -0,0 +1,182 @@
+    fn name(&self) -> &'static str {
+        "run-zkasm"
+    }
+
+    fn is_mutating(&self) -> bool {
+        false
+    }
+
+    fn needs_isa(&self) -> bool {
+        false
+    }
+
+    /// Runs the entire subtest for a given target, invokes [Self::run] for running
+    /// individual tests.
+    fn run_target<'a>(
+        &self,
+        testfile: &TestFile,
+        _: &mut FileUpdate,
+        _: &'a str,
+        _: &'a Flags,
+        _: Option<&'a dyn TargetIsa>,
+    ) -> anyhow::Result<()> {
+        // We can build the FunctionStore once and reuse it
+        let mut func_store = FunctionStore::default();
+        for (func, _) in &testfile.functions {
+            func_store.add(func.name.to_string(), &func);
+        }
+
+        for (func, details) in &testfile.functions {
+            info!("Test: {}({}) runner-zkasm", self.name(), func.name);
+
+            run_test(&func_store, func, details).context(self.name())?;
+        }
+
+        Ok(())
+    }
+
+    fn run(
+        &self,
+        _func: Cow<ir::Function>,
+        _context: &crate::subtest::Context,
+    ) -> anyhow::Result<()> {
+        unreachable!()
+    }
+}
+
+// TODO: replace by results of https://github.com/near/wasmtime/pull/213
+struct ZkAsmExecutor {}
+
+impl ZkAsmExecutor {
+    pub fn new() -> ZkAsmExecutor {
+        ZkAsmExecutor {}
+    }
+
+    pub fn execute(self, zkasm_function: &Vec<String>, args: &Vec<DataValue>) -> DataValue {
+        for line in zkasm_function {
+            println!("{}", line);
+        }
+        for arg in args {
+            println!("{:#?}", arg);
+        }
+        todo!();
+    }
+}
+
+fn run_test(func_store: &FunctionStore, func: &Function, details: &Details) -> anyhow::Result<()> {
+    let zkasm_function = generate_zkasm(func_store, func);
+    for comment in details.comments.iter() {
+        if let Some(command) = parse_run_command(comment.text, &func.signature)? {
+            trace!("Parsed run command: {}", command);
+            let _ = command.run(|_func_name, run_args| {
+                let mut args = Vec::with_capacity(run_args.len());
+                args.extend_from_slice(run_args);
+                let executor = ZkAsmExecutor::new();
+
+                // just wrap to make CI green until executor is not ready
+                // TODO: replace by actual execution and results checking
+                let _ = panic::catch_unwind(AssertUnwindSafe(|| {
+                    let _inner = executor.execute(&zkasm_function, &args);
+                }));
+                // let result = executor.execute(&zkasm_function, &args);
+                // Ok(vec![result])
+                Ok(Default::default())
+            });
+            // just to make CI green until runner is not ready
+            // TODO: remove this
+            // .map_err(|e| anyhow::anyhow!("{}", e))?;
+        }
+    }
+    Ok(())
+}
+
+fn generate_zkasm(_func_store: &FunctionStore, func: &Function) -> Vec<String> {
```

Comment:
I think we will remove ``test_zkasm.rs`` once ``test_run_zkasm.rs`` will be ready. And than do some refactoring, such as creating ``zkasm_codegen.rs`` if it will be needed.

Reply:
I agree that code duplication like this should be avoided. This code might be touched for upcoming refactorings or zkASM instrumentation. Keeping duplicate code in sync is tedious and if it gets out of sync that might lead to issues which are hard to debug.

Reply:
I'm definitely agree that we don't want have a code duplication. But, one more problem, it's not pure duplication, this functions are rewrited a bit for ``test_run_zkasm.rs``. And, until filetest infra is not ready we want to keep both versions -- one to keep existing test infra alive, the second one -- to make iterative progress to the new test infra. 

As alternative, I can suggest to create some ``filtest-dev`` branch. Code still will be reviewed before go to that branch, but we can make some exceptions for such problems

Reply:
Hopefully, https://github.com/near/wasmtime/pull/216 can be a good start for the refactoring to avoid duplication.

> But, one more problem, it's not pure duplication, this functions are rewrited a bit for test_run_zkasm.rs

Can you please document the necessary changes so that it is easier to reconcile these two functions in the future?